cli/cmd/compliance.go

```diff
@@ -435,3 +441,49 @@ func statusToProperTypes(status string) string {
 		return "Unknown"
 	}
 }
+
+func validateRecommendationID(s string) bool {
+	match, _ := regexp.MatchString(RecommendationIDRegex, s)
+	return match
+}
+
+func outputResourcesByRecommendationID(report api.CloudComplianceReport) error {
+	violations := filterResourcesByRecommendationID(report, compCmdState.RecommendationID)
+
+	if len(violations) == 0 {
+		cli.OutputHuman("\nNo resources found affected by '%s'\n", compCmdState.RecommendationID)
+		return nil
+	}
+
+	if cli.JSONOutput() {
+		resourcesJsonOut := struct {
+			Violations []api.ComplianceViolation `json:"violations"`
+		}{violations}
+		return cli.OutputJSON(resourcesJsonOut)
+	}
+
+	cli.OutputHuman(
+		renderSimpleTable(
+			[]string{"Resource", "Region", "Reasons"},
+			violationsToTable(violations),
+		),
+	)
+	cli.OutputHuman("\n%d resources showing affected by `%s`\n", len(violations), compCmdState.RecommendationID)
+	return nil
+}
+
+func filterResourcesByRecommendationID(report api.CloudComplianceReport, recID string) (violations []api.ComplianceViolation) {
+	for _, r := range report.GetComplianceRecommendations() {
+		if r.RecID == recID {
+			violations = append(violations, r.Violations...)
+		}
+	}
+	return
+}
```

Comment:
I don't think we are "filtering" but instead, we are just accessing one recommendation, I think we should add a function to the `api` Go package that gets you that information, something like:
```go
func (aws ComplianceAwsReport) GetComplianceRecommendation(recID string) (ComplianceRecommendation, bool) {
	for _, r := aws.Recommendations {
		if r.RecID == recID {
			return r, true
		}
	}
	return ComplianceRecommendation{}, false
}
```